pkg/provisioner/ironic/ironic.go

```diff
@@ -510,6 +556,25 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 			)
 		}
 
+		// When node exists but has no assigned port to it by Ironic and actuall address (MAC) is present
+		// in host config and is not allocated to different node lets try to create port for this node.
+		if p.bootMACAddress != "" && !p.nodeHasAssignedPort(ironicNode) && !p.isPortAllocatedToAddress(p.bootMACAddress) {
+			enable := true
+			p.log.Info("creating port for node in ironic", "MAC",
+				p.bootMACAddress)
+			_, err = ports.Create(
+				p.client,
+				ports.CreateOpts{
+					NodeUUID:   ironicNode.UUID,
+					Address:    p.bootMACAddress,
+					PXEEnabled: &enable,
+				}).Extract()
+			if err != nil {
+				result, err = transientError(errors.Wrap(err, "failed to create port in ironic"))
+				return
+			}
+		}
```

Comment:
Since this whole block is copied and pasted from the one at line 525, can we split it out into a private method? Or delete that one and move this after the if/else.

Reply:
moved to private method, seems more readable that way